pkg/cloud/gcp/actuators/machine/reconciler.go

```diff
@@ -278,3 +294,66 @@ func isNotFoundError(err error) bool {
 	}
 	return false
 }
+
+func fmtInstanceSelfLink(project, zone, name string) string {
+	return fmt.Sprintf(instanceLinkFmt, project, zone, name)
+}
+
+func (r *Reconciler) instanceExistsInPool(instanceLink string, pool string) (bool, error) {
+	// Get target pool
+	tp, err := r.computeService.TargetPoolsGet(r.projectID, r.providerSpec.Region, pool)
+	if err != nil {
+		return false, fmt.Errorf("unable to get targetpool: %v", err)
+	}
+
+	// Check if instance is in target pool already
+	if tp.Instances != nil {
+		for _, link := range tp.Instances {
+			if instanceLink == link {
+				return true, nil
+			}
+		}
+	}
+	return false, nil
+}
+
+type poolProcessor func(instanceLink, pool string) error
+
+func (r *Reconciler) processTargetPools(desired bool, poolFunc poolProcessor) error {
+	if r.providerSpec.TargetPools != nil && len(r.providerSpec.TargetPools) > 0 {
```

Comment:
nit; return early if this condition does not hold true. will help reduce the indentation.